File: BlackJackPython/.idea/BlackJack.py
import array
import random


class Cards:
    def getTotal(self,card,previoustotal):
        value=0
        total=previoustotal
        Totalarr =card.split(" of")
        match Totalarr[0]:
            case "Jack":
                value=11;
            case "Queen":
                value=12;
            case "King":
                value=13;
            case "Ace":
                value=1;
            case default:
                value=int(Totalarr[0])
        total=previoustotal+value
        return total
    def getCard(self,cardlist,start,PreviousList,PreviousTotal):
        total=PreviousTotal
        NewCardList=PreviousList
        if start:
            for i in range(0,2):
                index=random.randrange(0,len(cardlist))
                card=cardlist[index]
                total=self.getTotal(card,total)
                NewCardList.append(card);
        elif not start:
            index=random.randrange(0,len(cardlist))
            card=cardlist[index]
            total=self.getTotal(card,total)
            NewCardList.append(card);
        print("--------------------")
        for i in range(len(NewCardList)):
            print(NewCardList[i])
        print("")
        print("The Total Points are:",total)
        self.total=total
        return NewCardList
    def getTotalValue(self):
        return self.total

class Decks:
    def shuffle(self):
        ranks = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"];
        suits = ["Clubs", "Hearts", "Diamonds", "Spades"];
        DeckOfCards = []
        for i in range(len(suits)):
            for j in range(len(ranks)):
                cardValues = ranks[j] + " of " + suits[i]
                DeckOfCards.append(cardValues)
        return DeckOfCards;

class BlackJack:
    def main():
        option1='y'
        option2='y'
        cardList=[]
        DealerTrigger=True
        ReturnCardList=[]
        UpdatedTotal=0
        playerTotal=0
        DealerTotal=0
        start=True
        DecksObj=Decks()
        cardlist=DecksObj.shuffle()
        CardsObj=Cards()
        while True:
            while True:
                print("Player Side")
                ReturnCardList=CardsObj.getCard(cardlist,start,ReturnCardList,UpdatedTotal)
                UpdatedTotal=CardsObj.getTotalValue()
                start=False
                if (UpdatedTotal <= 21):
                    option2=input("Do You Want To hit card again y/n")
                    if(option2 == 'n'):
                        break
                else:
                    print("Dealer Won The Game")
                    option2='n'
                    DealerTrigger=False
                    break
            if(option2 == 'n' and DealerTrigger == True):
                playerTotal=UpdatedTotal
                start=True
                ReturnCardList.clear();
                print("Dealer Side")
                UpdatedTotal=0
                while(UpdatedTotal <=21 and DealerTotal<=16):
                    if(playerTotal !=DealerTotal):
                        ReturnCardList=CardsObj.getCard(cardlist,start,ReturnCardList,UpdatedTotal)
                        UpdatedTotal=CardsObj.getTotalValue()
                        start=False
                        DealerTotal=UpdatedTotal
                    elif(playerTotal == DealerTotal):
                        DealerTotal=0
                        playerTotal=0
                        print("Your Total:"+playerTotal+" !Dealer Total:"+DealerTotal+" DRAW")

                if(playerTotal > DealerTotal and playerTotal <= 21):
                    print("Your is BlackJack! Congrats")
                elif(DealerTotal > playerTotal and DealerTotal <= 21):
                    print("Dealer is BlackJack!You Lose")
                elif(playerTotal < DealerTotal and DealerTotal > 21):
                    print("Your is BlackJack! Congrats")
                elif(DealerTotal < playerTotal and playerTotal > 21):
                    println("Dealer is BlackJack!You Lose")
                elif (DealerTotal == playerTotal):
                        print("Dealer "+DealerTotal+" You "+playerTotal+"!Draw.");
            option1=input("Do You want to Play Again y/n")
            if(option1 == 'y'):
                ReturnCardList.clear()
                start=True
                DealerTrigger=True
                UpdatedTotal=0
                playerTotal=0
                DealerTotal=0
            elif(option1 == 'n'):
                break


if __name__ == "__main__":
    bj=BlackJack
    bj.main()

File: BlackJack/src/main/java/BlackJack.java
import java.util.*;
public class BlackJack
{
    public static List<String> ReturnCardList= new ArrayList<>();
    public static char option1 ='y',option2='y';
    public static boolean start=true,DealerTrigger=true;
    public static int UpdatedTotal=0,PlayerTotal=0,DealerTotal=0;
    public static void main(String[] args)
    {
        Scanner s=new Scanner(System.in);
        decks d=new decks();
        d.shuffle();
        List<String> list = d.getList();
        Cards c=new Cards();
        do{
            do
            {
                System.out.println("Player Side:");
                ReturnCardList=c.CardValues(list,start,ReturnCardList,UpdatedTotal);
                UpdatedTotal=c.GetCardsTotal();
                start=false;
                if(UpdatedTotal <=21)
                {
                    System.out.println("Do you want to hit card again y/n");
                    option2=s.next().charAt(0);
                    option2=Character.toLowerCase(option2);
                }
                else
                {
                    System.out.println("Dealer Won the game");
                    option2='n';
                    DealerTrigger=false;
                }
            }while(option2 == 'y');

            if(option2 == 'n' && DealerTrigger)
            {
                PlayerTotal=UpdatedTotal;
                start=true;
                ReturnCardList.clear();
                System.out.println("Dealer Side:");
                UpdatedTotal=0;
                while(UpdatedTotal <=21 && DealerTotal<=16)
                {
                    if(PlayerTotal != DealerTotal)
                    {
                        ReturnCardList=c.CardValues(list,start,ReturnCardList,UpdatedTotal);
                        UpdatedTotal=c.GetCardsTotal();
                        start=false;
                        DealerTotal=UpdatedTotal;
                    }
                    else if(PlayerTotal == DealerTotal)
                    {
                        DealerTotal=0;
                        PlayerTotal=0;
                        System.out.println("Your Total: "+PlayerTotal+" !Delare Total: "+DealerTotal+" DRAW");
                    }

                }
                if(PlayerTotal > DealerTotal && PlayerTotal <= 21)
                {
                    System.out.println("Your is BlackJack! Congrats");
                }
                else if(DealerTotal > PlayerTotal && DealerTotal <= 21)
                {
                    System.out.println("Dealer is BlackJack!You Lose");
                }
                else if(PlayerTotal < DealerTotal && DealerTotal > 21)
                {
                    System.out.println("Your is BlackJack! Congrats");
                }
                else if(DealerTotal < PlayerTotal && PlayerTotal > 21)
                {
                    System.out.println("Dealer is BlackJack!You Lose");
                } else if (DealerTotal == PlayerTotal)
                {
                    System.out.println("Dealer "+DealerTotal+" You "+PlayerTotal+"!Draw.");
                }
            }
            System.out.println("Do you want to Play Again y/n");
            option1=s.next().charAt(0);
            option1=Character.toLowerCase(option1);
            if(option1 == 'y')
            {
                ResetGame();
            }
        }while(option1 =='y');
    }
    public static void ResetGame()
    {
        ReturnCardList.clear();
        start=true;
        UpdatedTotal=0;
        PlayerTotal=0;
        DealerTotal=0;
        DealerTrigger=true;
    }
}
class Cards extends decks
{
    public List<String> NewCardList=new ArrayList<String>();
    public int total=0;
    public List<String> CardValues(List<String> CardList,boolean start,List<String> PreviousList,int PreviousTotal)
    {
        NewCardList=PreviousList;
        total=PreviousTotal;
        Random r=new Random();
        String card="";
        if(start)
        {
            for(int i=0;i<2;i++)
            {
                int index=r.nextInt(CardList.size());
                card= CardList.get(index);
                gettotal(card);
                NewCardList.add(card);
            }
        }
        else if(!start)
        {
            int index=r.nextInt(CardList.size());
            card= CardList.get(index);
            NewCardList.add(card);
            gettotal(card);
        }
        System.out.println("-----------------------");
        for(int i=0;i<NewCardList.size();i++)
        {
            System.out.println(NewCardList.get(i));
        }
        System.out.println();
        System.out.println("The Total Points are:"+total);
        return NewCardList;
    }
    public int GetCardsTotal()
    {
        return this.total;
    }
    public int gettotal(String card)
    {
        int value=0;
        String[] Totalarr=card.split(" of");
        switch(Totalarr[0])
        {
            case "Jack":
                value=11;
                break;
            case "Queen":
                value=12;
                break;
            case "King":
                value=13;
                break;
            case "Ace":
                value=1;
                break;
            default:value= Integer.parseInt(Totalarr[0]);
        }
        total=total+value;
        return total;
    }
}
class decks
{
    public String[] ranks = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King","Ace"};
    public String[] suits ={"Clubs","Hearts","Spades","Diamonds"};
    public ArrayList<String> DecksofCards=new ArrayList<String>();
    public void shuffle()
    {
        String CardValue;
        for(int i=0;i<suits.length;i++)
        {
            for(int j=0;j<ranks.length;j++)
            {
                CardValue=ranks[j]+" of "+suits[i];
                DecksofCards.add(CardValue);
            }
        }
        setList(DecksofCards);
    }
    public void setList(ArrayList list) {
        this.DecksofCards = list;
    }
    public ArrayList getList() {
        return DecksofCards;
    }
}